modules/table/src/integrationTest/java/org/apache/ignite/internal/table/distributed/disaster/ItHighAvailablePartitionsRecoveryTest.java

```diff
@@ -297,6 +302,62 @@ void testHaZoneScaleDownNodesDoNotRemovedFromStable() throws InterruptedExceptio
         waitAndAssertStableAssignmentsOfPartitionEqualTo(node, HA_TABLE_NAME, PARTITION_IDS, allNodes);
     }
 
+    @Test
+    void testRebalanceInHaZone() throws InterruptedException {
+        createHaZoneWithTable();
+
+        startNode(3);
+
+        Set<String> fourNodes = runningNodes().map(Ignite::name).collect(Collectors.toUnmodifiableSet());
+
+        executeSql(format("ALTER ZONE %s SET REPLICAS=%d", HA_ZONE_NAME, 4));
+
+        IgniteImpl node = igniteImpl(0);
+
+        int tableId = getTableId(node.catalogManager(), HA_TABLE_NAME, clock.nowLong());
+
+        for (Integer partId : PARTITION_IDS) {
+            assertRaftGroupsOnNodes(node, partId, tableId, HA_ZONE_NAME, 0, 1, 2, 3);
```

Comment:
AFAIK, this is "Verify that the partition is rebalanced to include node D" step.
Why can't we just check that D is in stable?